mautic/mautic
# Implement custom bounties workflow with Open Collective

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [✅  ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] N/A
| Related user documentation PR URL      | https://github.com/mautic/mautic-community-handbook/pull/175/files
| Related developer documentation PR URL | N/A
| Issue(s) addressed                     | Addresses this debate: [https://community.mautic.org/assemblies/product-team/f/39/debates/2](https://community.mautic.org/assemblies/product-team/f/39/debates/2)


#### Description:

Bountysource has [stopped paying out to developers](https://github.com/bountysource/core/issues/1586#issuecomment-1703675937) so we need to change the way our bounties work.

After [some debate](https://community.mautic.org/assemblies/product-team/f/39/debates/2), folks felt that it would be preferential to have a system which uses Open Collective and no other third party, so that we don't attract unnecessary fees. This might be a bit clunky to start with, but it gets us going with something which will hopefully start the ball rolling and can be iterated on.

This PR adds a workflow for GitHub Actions which:

-  Adds text on the bottom of any new issue created, advising people that if they are a member (see point 1 below) they can add funds to the bounty pot which can be assigned to the issue.
- Posts a message when the Bounty label is assigned, informing anyone watching (and later viewers) that there is a bounty associated with the issue.

Potential future developments:
- Find a way to identify the donations for an issue (maybe create a specific tier via the API when the Bounty label is added?) so we can programmatically update the title / messages with the amount available
- Automatically post instructions on how to claim the bounty when the corresponding PR is merged

1. I think it makes sense to make bounties a privilege of membership but open to thoughts on that - please see the debate linked above.

#### Steps to test this PR:
This can't easily be tested but you can check my fork here: https://github.com/RCheesley/mautic/issues/26 which has the text appended and the message about a bounty being available. The title is a manual process, I suggest we get the person adding funds to tell us how much so we can add them.
